src/components/styledSelect/styledSelect.tsx

```diff
@@ -0,0 +1,147 @@
+
+const StyledSelect = ({
+  options,
+  placeholder,
+  onOptionSelected,
+  defaultOption,
+  disabled,
+  className,
+}: StyledSelectProps) => {
+  const [activeOption, setActiveOption] = useState<SelectOption | null>(null)
+  const [headerText, setHeaderText] = useState(defaultOption?.optionName || '')
+  const [expanded, setExpanded] = useState(false)
+  const componentRef = useRef<HTMLDivElement>(null)
+
+  const colorVars = {
+    '--button-background-color': BLUE.schemeColorDarkest,
+    '--button-hover-color': BLUE.hoverColorDark,
+    '--button-text-color': BLUE.textColorPrimary,
+    '--button-border-color': BLUE.borderColor,
+  } as CSSProperties
+
+  const toggleDropdown: MouseEventHandler = (e) => {
+    e.preventDefault()
+
+    if (disabled) return
+
+    setExpanded(!expanded)
+  }
+
+  const selectOption = (value: string | number) => {
+    const opt = options.find(({ optionValue }) => optionValue === value)
+
+    if (opt) {
+      setActiveOption(opt)
+      setHeaderText(opt.optionName)
+      setExpanded(false)
+      onOptionSelected(value)
+    }
+  }
+
+  useEffect(() => {
+    const hideDropdown = (e: FocusEvent) => {
+      if (
+        componentRef.current &&
+        componentRef.current !== e.relatedTarget &&
+        !componentRef.current.contains(e.relatedTarget as Node)
+      ) {
+        setExpanded(false)
+      }
+    }
+
+    document.addEventListener('focusout', hideDropdown)
+
+    return () => document.removeEventListener('focusout', hideDropdown)
+  })
+
+  useEffect(() => {
+    if (defaultOption && !activeOption) {
+      setHeaderText(defaultOption.optionName)
+      setActiveOption(defaultOption)
+    }
+  }, [defaultOption, activeOption])
+
+  return (
+    <div
+      className={classNames(styles.root, className, {
+        [styles.disabled]: disabled,
+      })}
+      style={colorVars}
+      ref={componentRef}
+      data-testid="styledSelect"
+    >
+      <div
+        className={styles.header}
+        role="combobox"
+        aria-haspopup="listbox"
+        aria-owns="selectListbox"
+        aria-controls="selectListbox"
+        aria-expanded={expanded}
+        onClick={toggleDropdown}
+      >
+        <p className={styles.headerText} data-testid="selectedOption">
+          {truncatedText(options.length ? headerText : placeholder)}
+        </p>
+        <button className={styles.trigger} disabled={disabled}>
+          <FontAwesomeIcon className={styles.fa} icon={faAngleDown} />
+        </button>
+      </div>
+      <ul
+        id="selectListbox"
+        className={classNames(styles.dropdown, {
+          [styles.hidden]: !expanded || !options.length,
+        })}
+        role="listbox"
+      >
+        {options.map((option, index) => {
+          return (
+            <StyledSelectOption
+              key={index}
+              onSelected={selectOption}
+              ariaSelected={!!activeOption && isEqual(option, activeOption)}
+              optionName={option.optionName}
```

Comment:
This was introduced in #45, where I had planned to remove it because it didn't end up contributing to the bugfix. However, I ended up merging the PR before I pushed my local changes by mistake. We should change this to the way it was before:
```
{...option}
```